pkg/util/kubernetes/cloudprovider/cloudprovider.go

```diff
@@ -0,0 +1,78 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package cloudprovider
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config/setup/constants"
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+	"github.com/DataDog/datadog-agent/pkg/util/clusteragent"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/hostinfo"
+)
+
+// GetName returns the name of the cloud provider for the current node.
+// GetName shouldn't be used on DCA. For DCA please refer to DCAGetName.
+func GetName() (string, error) {
+	cacheKey := cache.BuildAgentKey(constants.NodeCloudProviderKey)
+	if cloudProvider, found := cache.Cache.Get(cacheKey); found {
+		return cloudProvider.(string), nil
+	}
+
+	ni, err := hostinfo.NewNodeInfo()
+	if err != nil {
+		return "", err
+	}
+	ctx, cancel := context.WithTimeout(context.TODO(), 2*time.Second)
```

Comment:
I forgot to mention we should use `context.Background()` instead of `.TODO()`.

Reply:
Any rational ?